### ISSUE TITLE
Convert all values to unicode in models.py

### ISSUE THREAD

**Issue by cornelinux** [privacyidea/privacyidea#1000]
Just like with the issue #999 we should convert all input data to unicode in ``models.py``, if the columns are defined as unicode.
This way we can assure, that if some settings provide e.g. an integer, the database receives the correct data.

    self.Key = unicode(Key)
    self.Value = unicode(Value)

We need to adapt several tests, since the tests often test for ``==1``, since sqlite converts bool "True" to integer "1". We need to test with ``is_true``.

**Comment by fredreichbier** [comment]
A problem I just stumbled upon: In some cases, e.g. the RADIUS token, ``Value`` can actually be ``None``. Up until now, I think we then stored ``NULL`` in the databsae. If we now use ``unicode(Value)``, we will store ``"None"`` (the string) in the database instead. I guess we should do something like
```python
if Value is None:
    self.Value = Value
else:
    self.Value = unicode(Value)
```
What do you think?

**Comment by cornelinux** [comment]
I am totally find with storing ``None``. And I would expect that oracle could store ``None`` in a varchar field.
So I am fine with your ``if-else``-approach above.

**Comment by fredreichbier** [comment]
reopening this because I forgot the ``Audit`` table

**Comment by fredreichbier** [comment]
We can already see the bug if we use SQLite. If we create a policy and choose at least one realm, the ``audit_data`` dictionary contains ``'realm': [u"defrealm"]``, i.e., a real Python dictionary. Because of that, the entry is not added to the audit log:
```
[2018-05-04 13:50:43,295][5745][140498025260224][ERROR][privacyidea.lib.auditmodules.sqlaudit:240] exception InterfaceError('(sqlite3.InterfaceError) Error binding parameter 7 - probably unsupported type.',)
[2018-05-04 13:50:43,296][5745][140498025260224][ERROR][privacyidea.lib.auditmodules.sqlaudit:241] DATA: {'info': u"{u'realm': [u'defrealm'], u'check_all_resolvers': False, u'action': [u'passOnNoToken'], u'client': [], u'user': [], u'resolver': u'', u'time': u'', u'active': True, u'scope': u'authentication', u'adminrealm': []}", 'administrator': u'admin', 'realm': [u'defrealm'], 'success': True, 'privacyidea_server': 'localhost:5000', 'client_user_agent': 'firefox', 'token_type': None, 'client': '127.0.0.1', 'user': '', 'resolver': '', 'action_detail': u'nouser', 'action': 'POST /policy/<name>', 'serial': None}
```

**Comment by fredreichbier** [comment]
I believe we have finally fixed this by applying #1047.
